### DISCUSSION TITLE
Unpublish pages / 404 pages / Markdown Drafts

### DISCUSSION THREAD

**Discussion by courcelan** [ElMassimo/iles#73] [Q&A] [answered]
Is there a way to force a page to not be published? Such as in `extendFrontmatter` or `extendRoute`?

Also, is there a way to set up a 404 page?

Thanks!

**Comment by ElMassimo** [comment, marked as answer]
Hi Cobey!

 For setting up a 404 page, you can create a [`404` page](https://github.com/ElMassimo/iles/blob/main/docs/src/pages/404.vue) with any of the supported extensions (`md`, `mdx`, `vue`, `html`), or use `path: /404.html` in the frontmatter.

Service providers such as Netlify and Cloudflare Pages will pick it up automatically and use it for [unresolved html requests](https://iles-docs.netlify.app/cobey). 

---

To "unpublish" a page, you can use the [_Drafts_](https://iles.pages.dev/guide/markdown#markdown-drafts) feature.

```mdx
---
draft: true
---
```

---

If you need more control, you can remove it from the generated routes using [`extendRoutes`](https://iles-docs.netlify.app/guide/plugins#islandspages):

```js
  extendRoutes (routes) {
    return routes.filter(someCondition)
  },
```

**Reply by ElMassimo** [reply]
@courcelan Updated to mention the new [_Drafts_](https://iles.pages.dev/guide/markdown#markdown-drafts) feature.
